plugins/module_utils/kibana.py

```diff
@@ -70,4 +71,210 @@ def get_alert_connector_by_name(self, connector_name):
+            return(rule['name'] + ": Rule is updated")
+          elif rule['enabled'] == True and rule_name == rule['name']:
+            return(rule['name'] + ": Rule is already enabled")
+        page_number = page_number + 1
+        rules = rule_actions.get_rules(page_size,page_number)
+        allrules = rules['data']
+    return(rule['name'] + ": Rule not found")
+
+  # Elastic Integration functions
+
+  def get_integrations(self):
+      endpoint  = 'fleet/epm/packages'
+      integration_objects = self.send_api_request(endpoint, 'GET')
+      return integration_objects
+  
+  def install_integration(self, integration_name, version):
+      body = {
+          "force": True
+      }
+      body_JSON = dumps(body)
+      endpoint  = 'fleet/epm/packages/' + integration_name + "-" + version
+      if not self.module.check_mode:
+        integration_install = self.send_api_request(endpoint, 'POST', data=body_JSON)
+      else:
+        integration_install = "Cannot proceed with check_mode set to " + self.module.check_mode
+      return integration_install
+  
+  def check_integration(self, integration_name):
+      integration_action = Kibana(self.module)
+      integration_objects = integration_action.get_integrations()
+      integration_object = ""
+      for integration in integration_objects['response']:
+        if integration['title'] in integration_name:
+          integration_object = integration
+          if integration['status'] != 'installed':
+            integration_install = integration_action.install_integration(integration['name'],integration['version'])
+      return(integration_object)
+
+  # Elastic Integration Package Policy functions
+
+  def get_all_pkg_policies(self):
+      endpoint  = 'fleet/package_policies'
+      pkgpolicy_objects = self.send_api_request(endpoint, 'GET')
+      return pkgpolicy_objects
+  
+  def update_pkg_policy(self,pkgpolicy_id,body):
+      endpoint = "fleet/package_policies/" + pkgpolicy_id
+      if not self.module.check_mode:
+        pkg_policy_update = self.send_api_request(endpoint, 'PUT', data=body)
+      else:
+        pkg_policy_update = "Cannot proceed with check_mode set to " + self.module.check_mode
+      return pkg_policy_update
+  
+  def get_pkg_policy(self,agent_policy_id):
+    self.integration_name = self.module.params.get('integration_name')
+    pkg_policy_action = Kibana(self.module)
+    pkg_policy_objects = pkg_policy_action.get_all_pkg_policies()
+    pkg_policy_object = ""
+    for pkgPolicy in pkg_policy_objects['items']:
+      if pkgPolicy['package']['title'] == self.integration_name and pkgPolicy['policy_id'] == agent_policy_id:
+        pkg_policy_object = pkgPolicy
+    return(pkg_policy_object)
+  
+  def create_pkg_policy(self,agent_policy_id, integration_object):
+    self.pkg_policy_name = self.module.params.get('pkg_policy_name')
+    self.pkg_policy_desc = self.module.params.get('pkg_policy_desc')
```

Comment:
Module parameters should be gathered in the module or in `init` if in module_utils as mentioned above.